subsys/net/lib/nrf_cloud/src/nrf_cloud_codec.c

```diff
@@ -1740,19 +1762,86 @@ int nrf_cloud_format_cell_pos_req_json(struct lte_lc_cells_info const *const inf
 	return -ENOMEM;
 }
 
-int nrf_cloud_format_cell_pos_req(struct lte_lc_cells_info const *const inf,
-	size_t inf_cnt, char **string_out)
+int nrf_cloud_format_wifi_req_json(struct wifi_scan_info const *const wifi,
+	cJSON *const req_obj_out)
 {
-	if (!inf || !inf_cnt || !string_out) {
+	if (!wifi || !req_obj_out || !wifi->ap_info || !wifi->cnt) {
+		return -EINVAL;
+	}
+
+	cJSON *wifi_obj = NULL;
+	cJSON *ap_array = NULL;
+
+	wifi_obj = cJSON_AddObjectToObjectCS(req_obj_out, NRF_CLOUD_LOCATION_JSON_KEY_WIFI);
+	ap_array = cJSON_AddArrayToObjectCS(wifi_obj, NRF_CLOUD_LOCATION_JSON_KEY_APS);
+	if (!ap_array) {
+		goto cleanup;
+	}
+
+	for (uint8_t cnt = 0; cnt < wifi->cnt; ++cnt) {
+		struct wifi_ap_info const *const ap = (wifi->ap_info + cnt);
+		cJSON *ap_obj = cJSON_CreateObject();
+
+		if (!cJSON_AddItemToArray(ap_array, ap_obj)) {
+			cJSON_Delete(ap_obj);
+			goto cleanup;
+		}
+
+		/* required items */
+		if (json_add_str_cs(ap_obj, NRF_CLOUD_LOCATION_JSON_KEY_WIFI_MAC,
+				    ap->mac_addr_str)) {
+			goto cleanup;
+		}
+
+		/* optional */
+		if ((ap->ssid_str[0] != '\0') &&
+		    json_add_str_cs(ap_obj, NRF_CLOUD_LOCATION_JSON_KEY_WIFI_SSID,
+				    ap->ssid_str)) {
+			goto cleanup;
+		}
+		if ((ap->rssi != NRF_CLOUD_LOCATION_WIFI_OMIT_RSSI) &&
+		    json_add_num_cs(ap_obj, NRF_CLOUD_LOCATION_JSON_KEY_WIFI_RSSI,
+					ap->rssi)) {
+			goto cleanup;
+		}
+		if ((ap->rssi != NRF_CLOUD_LOCATION_WIFI_OMIT_CHAN) &&
+		    json_add_num_cs(ap_obj, NRF_CLOUD_LOCATION_JSON_KEY_WIFI_CH,
+					ap->channel)) {
+			goto cleanup;
+		}
+	}
+
+	return 0;
+
+cleanup:
+	/* Only need to delete the WiFi object since all items (if any) were added to it */
+	cJSON_DeleteItemFromObject(req_obj_out, NRF_CLOUD_LOCATION_JSON_KEY_WIFI);
+	LOG_ERR("Failed to format WiFi location request, out of memory");
+	return -ENOMEM;
+}
+
+int nrf_cloud_format_location_req(struct lte_lc_cells_info const *const cell_info,
+	struct wifi_scan_info const *const wifi_info, char **string_out)
+{
+	if ((!cell_info && !wifi_info) || !string_out) {
 		return -EINVAL;
 	}
 
 	int err = 0;
 	cJSON *req_obj = cJSON_CreateObject();
 
-	err = nrf_cloud_format_cell_pos_req_json(inf, inf_cnt, req_obj);
-	if (err) {
-		goto cleanup;
+	if (cell_info) {
+		err = nrf_cloud_format_cell_pos_req_json(cell_info, 1, req_obj);
```

Comment:
Just wondering whether the `inf_cnt` parameter is still needed? This function is called from two places and both set the count to 1, I think.

Reply:
multiple object are accepted by the API and i think that feature will be used at some point.

Reply:
I think that `inf_cnt` will not be needed. At least, with new GCI search types as those are also included within lte_lc_cells_info, where are all included: current/serving cell with it's neighbors + surrounding GCI cells found by GCI search types. Thus, at least with this this data structure there isn't need for passing array of cell_infos.

Reply:
Ah ok, thanks.
I would like to save that change for a separate PR.

Reply:
created `IRIS-5078` for this